src/transformers/pipelines/pt_utils.py

```diff
@@ -111,6 +111,44 @@ def loader_batch_item(self):
             # Recreate the element by reusing the original class to make it look
             # batch_size=1
             result = self._loader_batch_data.__class__(loader_batched)
+        else:
+            # Batch data is assumed to be tuple
+            loader_batched = []
+            for element in self._loader_batch_data:
```

Comment:
AFAICT -  almost all of this code is a copy-paste of the logic above and so should be abstracted out.